meson.build

```diff
@@ -16,16 +16,31 @@ project(
 # for a reference about clean library versioning.
 so_version = ['1', '2', '0']
 
-add_project_arguments(
-	[
-		'-DWLR_SRC_DIR="@0@"'.format(meson.current_source_dir()),
-		'-DWLR_USE_UNSTABLE',
+add_project_arguments([
+	'-DWLR_SRC_DIR="@0@"'.format(meson.current_source_dir()),
+	'-DWLR_USE_UNSTABLE',
+], language: 'c')
 
-		'-Wno-unused-parameter',
-		'-Wundef',
-	],
-	language: 'c',
-)
+cc = meson.get_compiler('c')
+
+add_project_arguments(cc.get_supported_arguments([
+	'-Wundef',
+	'-Wlogical-op',
+	'-Wmissing-include-dirs',
+	'-Wold-style-definition',
+	'-Wpointer-arith',
+	'-Winit-self',
+	'-Wstrict-prototypes',
+	'-Wredundant-decls',
+	'-Wimplicit-fallthrough=2',
+	'-Wendif-labels',
+	'-Wstrict-aliasing=2',
+	'-Woverflow',
+
+	'-Wno-missing-braces',
+	'-Wno-missing-field-initializers',
+	'-Wno-unused-parameter',
+]), language: 'c')
```

Comment:
Can we summarize this as `-Wall -Wextra -Wpedantic`?

Reply:
No, `-Wpedantic` gives a lot of these:

```
../types/wlr_output_layout.c: In function ‘wlr_output_layout_add’:
../types/wlr_output_layout.c:211:66: warning: ISO C99 requires at least one argument for the "..." in a variadic macro
    wlr_log(WLR_ERROR, "Failed to create wlr_output_layout_output");
```

And there's no way to disable this specific warning. There's also no easy way to fix `wlr_log`.

Reply:
Also, meson already control those as as `warning_level` option.

Reply:
Yeah, the idea was to bump `warning_level` to 3 to enable `-Wpedantic`